lib/migrations/2_deploy_schemes.ts

```diff
@@ -0,0 +1,98 @@
+    const ExecutableTest = artifacts.require("ExecutableTest.sol");
+    const GenesisProtocol = artifacts.require("GenesisProtocol.sol");
+    const GlobalConstraintRegistrar = artifacts.require("GlobalConstraintRegistrar.sol");
+    const SchemeRegistrar = artifacts.require("SchemeRegistrar.sol");
+    const SimpleICO = artifacts.require("SimpleICO.sol");
+    const TokenCapGC = artifacts.require("TokenCapGC.sol");
+    const UpgradeScheme = artifacts.require("UpgradeScheme.sol");
+    const VestingScheme = artifacts.require("VestingScheme.sol");
+    const VoteInOrganizationScheme = artifacts.require("VoteInOrganizationScheme.sol");
+    const UController = artifacts.require("UController.sol");
+
+    console.log(`Deploying schemes to ${network}`);
+
+    const DAOToken = await Utils.requireContract("DAOToken");
+    const gasLimit = await computeMaxGasLimit(web3);
+    let genTokenAddress;
+
+    await DAOToken.at("0x543Ff227F64Aa17eA132Bf9886cAb5DB55DCAddf")
+      .then((token: any): void => genTokenAddress = token.address)
+      .catch(() => { console.log(`global GEN token does not exist`); });
+
+    if (!genTokenAddress) {
+      if (network === "ganache") {
+        // then we will create a new token to use for staking
+        const genToken = await DAOToken.new("GEN", "GEN", web3.toWei(100000000));
+        genTokenAddress = genToken.address;
+        // and mint GEN to all of the accounts
+        const getAccounts = promisify(web3.eth.getAccounts);
+        const accounts = await getAccounts();
+        accounts.forEach((account: Address) => {
+          genToken.mint(account, web3.toWei(1000));
+        });
+      } else {
+        throw new Error(`The GEN token must exist for staking on ${network}`);
+      }
+    }
+
+    console.log(`Using global GEN token for staking on ${network} at: ${genTokenAddress}`);
+    console.log(`Deploying schemes on ${network}, gasLimit: ${gasLimit}`);
+
+    await deployer.deploy(ControllerCreator, { gas: gasLimit });
+    const controllerCreator = await ControllerCreator.deployed();
+    await deployer.deploy(DaoCreator, controllerCreator.address, { gas: gasLimit });
+    await deployer.deploy(UController, { gas: gasLimit });
+    await deployer.deploy(GenesisProtocol, genTokenAddress, { gas: gasLimit });
+    await deployer.deploy(SchemeRegistrar, { gas: gasLimit });
+    await deployer.deploy(UpgradeScheme, { gas: gasLimit });
+    await deployer.deploy(GlobalConstraintRegistrar, { gas: gasLimit });
+    await deployer.deploy(ContributionReward, { gas: gasLimit });
+    await deployer.deploy(AbsoluteVote, { gas: gasLimit });
+    await deployer.deploy(SimpleICO, { gas: gasLimit });
+    await deployer.deploy(TokenCapGC, { gas: gasLimit });
+    await deployer.deploy(VestingScheme, { gas: gasLimit });
+    await deployer.deploy(VoteInOrganizationScheme, { gas: gasLimit });
+    if (network !== "live") {
+      await deployer.deploy(ExecutableTest, { gas: gasLimit });
```

Comment:
why ExecutableTest need to be part of the deployment ? 
who is using that ?

Reply:
There is an Arc.js test that uses it.